stylix/autoload.nix

```
@@ -8,7 +8,20 @@ builtins.concatLists (
     path: kind:
     let
       file = "${inputs.self}/modules/${path}/${for}.nix";
+      module = import file;
+      # Assume that a function-module whose first arg doesn't have any de-
+      # structured attrs is actually expecting `mkTarget` as its first arg;
+      #
+      # `mkTarget` cannot be distribtued normally through the module system
+      # due to issues of infinite recursion
+      expectsMkTarget =
+        builtins.isFunction module && builtins.functionArgs module == { };
     in
-    lib.optional (kind == "directory" && builtins.pathExists file) file
+    lib.optional (kind == "directory" && builtins.pathExists file) (
+      if expectsMkTarget then
+        lib.modules.importApply file (import ./mk-target.nix)
```

I was just reading the docs again for `importApply` and noticed a detail I'd missed before:

> This function does not add support for deduplication and `disabledModules`,
> although that could be achieved by wrapping the returned module and setting
> the `_key` module attribute.
> 
> The reason for this omission is that the file path is not guaranteed to be
> a unique identifier for the module, as two instances of the module may
> reference different `arg`s in their closures.

This contradicts what I said in https://github.com/danth/stylix/pull/1185#issuecomment-2848319802 and https://github.com/danth/stylix/pull/1185#issuecomment-2849481858.

It is also the first I've heard of the `_key` attribute. It sounds similar to the `_file` attribute that _is_ set by `importApply`.

I'd need to look into `disabledModules` and `_key` a bit more, but perhaps the correct approach here is:


```suggestion
        {
          _key = file;
          _file = file;
          imports = [ (module mkTarget) ];
        }
```

?

---

This is a little above my pay grade... I'll put it in and we'll see CI's response.

---

Looking at the impl in `evalModules`, I think the `importApply` docs may be slightly off; I believe it is `key` not `_key`.